### ISSUE TITLE
Gallery : image quality enhancement features

### ISSUE THREAD

**Issue by gbdomubpkm** [HelloVolla/volla-os-beta-test#45]
What is the status of 'Gallery' on the Volla?
It says v 6.20.3 Pro Simple Mobile Tools but is it really? Is it a Pro version because many of the image quality improvement functions (light, contrast, saturation, shadows...) do not seem to be accessible.
Only pre-programmed effects seem to be available. Do i miss something ?
I would like to be able to access advanced functions? Is it possible?
Thanks

**Comment by wurzer** [comment]
I suggest to post your requirement on the project's page: https://github.com/SimpleMobileTools/Simple-Gallery. 
I can see a bunch of editing features:

![Screenshot_20210720-092518](https://user-images.githubusercontent.com/7789612/126279583-9e455c91-62e3-46d0-a2ea-d7123546bf32.png)

**Comment by gbdomubpkm** [comment]
Yes indeed me the same for these functions but are not accessible (or I do not find?) those that I describe you.

**Comment by wurzer** [comment]
You tap on the pencil button after you have tapped on on a photo.
![Screenshot_20210720-123921](https://user-images.githubusercontent.com/7789612/126310772-02d4b11a-0419-45dd-91cf-6440c1cbcd69.png)

**Comment by gbdomubpkm** [comment]
I speak of that 
![Image enhancement functions](https://user-images.githubusercontent.com/50362604/126319478-99618227-7b4e-4109-8e57-844f65e9c133.png)
Did you succeed ?

**Comment by wurzer** [comment]
This picture shown another app, doesn't it?

**Comment by gbdomubpkm** [comment]
No,  it is this app, Simple Gallery Pro v 6.20.3,  but picture shown using Aurora Store.
We can also show and open the same app using F-Droïd.
That is true that the pictures aren't exactly the same on F-Droïd aud Aurora Store, but the features i want to use can be found on F-Droïd in Adjust, Filter.. (Practical Editor  Photo).

i have send a mail to Simple Mobile Tools.

**Comment by gbdomubpkm** [comment]
I have got an answer. I'm not quite sure I understood what they told me, but i think you should contact with Simple Mobile Tools team about this app..

**Comment by wurzer** [comment]
It's because you need to unlock features with the Thank You App?

**Comment by gbdomubpkm** [comment]
Not only. About that, they said me it's not possible outside google play store if i well understood.
It's rather that they have questions about the use of their (cloned ?) app by Volla and I'm unable to answer them and it's not my role.

**Comment by wurzer** [comment]
The owner of the project knows us. We have no clone. It's quite simple. We use their distribution on F-Droid.

**Comment by gbdomubpkm** [comment]
Happy to ear that it's quite simple 🙂🙂.
The person who answered me by email made a wrong assumption about 'cloned'..
So just come back to the question of being able to unlock these image improvement features that I (maybe not just me) want to use..

**Comment by wurzer** [comment]
Di you try to install the simple "Thank You" app from F-Droid?

**Comment by gbdomubpkm** [comment]
Yes. Nothing changed for that. It is the advanced Editor (not free i think) that we can't use (currently ?).

**Comment by wurzer** [comment]
I assume that the images in the app store are not updated. They are shown for the free as well for the pro app.

**Comment by gbdomubpkm** [comment]
So it's unsolvable unless you can negociate a way with the owner of Gallery to include the pro version on the Volla which would be welcome.

**Comment by wurzer** [comment]
You will find the Gallery and Gallery Pro App in the Aurora and Google Play Store. We have already pre-installed the Gallery Pro app on the Volla Phone, hat is published under GPL. You can derive this from the package name in the settings.

**Comment by gbdomubpkm** [comment]
Yes it is,  indeed the name is Pro but the pro features are blocked..

**Comment by gbdomubpkm** [comment]
Following an advice I was given, I tried to install an apk from the source https://github.com/SimpleMobileTools/Simple-Gallery/releases which could (maybe) have solved the problem. Anyway, the process fails at the end because on the Volla, Gallery "Pro" is currently a built-in app (can't be deleted, only desabled. ) and an Github update can't be done over it : failure. Disabling Gallery and installing the Github source is a failure too.

About built-in app as Gallery, Will they stay integrated on Volla OS based on android 10 ? Or can they be uninstalled ?

**Comment by wurzer** [comment]
The build from Github is the one, that is published on F-Droid. However you can compare the apps, if you make your own build with a different package name. 

https://github.com/SimpleMobileTools/Simple-Gallery/blob/master/app/src/main/AndroidManifest.xml

Here is a list of views in the open source app:

https://github.com/SimpleMobileTools/Simple-Gallery/tree/master/app/src/main/res/layout

Android uses several partitions respectively images on the device's storage. Apps, that are deployed on the system image and are part of the operating system can only be de-activated. 

If you install apps on your own, they are installed on the user partition, where they are deleted, if you do a factory reset. This is an impact of the security approach. 

which features are you missing?  I can see the photo editor with a bunch of functions. .

**Comment by gbdomubpkm** [comment]
Installing  apk from Github source to have the advanced editor, works on a basic android phone: I just tested.

Install the latest Gallery Pro apk from the Github source on a basic android smartphone and open a picture with the advanced editor, you will see better what I would like to get and use :

But here are 2 screenshots of the two desired menus (which scroll and have other options on their right)

![IMG_20210723_211245_4_1](https://user-images.githubusercontent.com/50362604/126853478-a4c4a57b-8b09-44f2-bf86-ee6fe901b04b.jpg)
![IMG_20210723_211318_5_1](https://user-images.githubusercontent.com/50362604/126853480-ac3a8306-80ef-44a1-8bbd-547ce1cfd70c.jpg)

**Comment by wurzer** [comment]
Thanks for this report. We may include the APK from Github then in our additional repository for F-Droid

**Comment by gbdomubpkm** [comment]
What is Volla Nightly Repository in F-Droid ?
Should we activate this repository which is not activated by default?

**Comment by wurzer** [comment]
We have a nightly build repository. Go to the repository settings of F-Droid and you find the nightly build channel, that you can activate. Your idea is to provide the simple mobile app apk from Github right here? It could work.

**Comment by gbdomubpkm** [comment]
Is a solution planned ?

**Comment by wurzer** [comment]
We will evaluate the solution to use an own build or the build of Github.

**Comment by gbdomubpkm** [comment]
Any news ? Thanks in advance.

**Comment by gbdomubpkm** [comment]
I just go to info to know if there is something new about this feature request. Thanks.

**Comment by gbdomubpkm** [comment]
I tried to install last github apk on AOSP 11.1. Doesn't work.

**Comment by wurzer** [comment]
I see, It's not possible to install both apps next to each other, because they have the same package name. And for updating the pre-installed app, they have a different signature.
